assets/js/middleware/store-api-nonce.js

```diff
@@ -28,12 +37,44 @@ const isStoreApiGetRequest = ( options ) => {
  * @param {Object} headers Headers object.
  */
 const setNonce = ( headers ) => {
-	const newNonce = headers?.get( 'X-WC-Store-API-Nonce' );
-	if ( newNonce ) {
-		nonce = newNonce;
+	const nonce = headers?.get( 'X-WC-Store-API-Nonce' ) || '';
+	const timestamp = headers?.get( 'X-WC-Store-API-Nonce-Timestamp' ) || 0;
+
+	if ( nonce ) {
+		updateNonce( nonce, timestamp );
 	}
 };
 
+/**
+ * Updates the stored nonce within localStorage so it is persisted between page loads.
+ *
+ * @param {string} nonce Incoming nonce string.
+ * @param {number} timestamp Timestamp from server of nonce.
+ */
+const updateNonce = ( nonce, timestamp ) => {
+	// If the "new" nonce matches the current nonce, we don't need to update.
+	if ( nonce === currentNonce ) {
+		return;
+	}
+
+	// Only update the nonce if newer. It might be coming from cache.
+	if ( currentTimestamp && timestamp < currentTimestamp ) {
+		return;
+	}
+
+	currentNonce = nonce;
+	currentTimestamp = timestamp || Date.now() / 1000;
```

Comment:
Perhaps add a comment for why we're dividing by 1000 here, without looking at L68 of Assets.php it's not _immediately_ obvious. Not a big deal I think but may be helpful.

Reply:
Thanks, added in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3770/commits/7aa18669a36a7bdb3c5c91f4cc47a26869edef7c